fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,211 @@
+	# the list ``sceneNames`` needs to be synced with ``fileData``
+	fileData, header, sceneNames = getSceneTable(exportPath)
+	sceneName, sceneTitle, sceneID, sceneUnk10, sceneUnk12, sceneIndex = getSceneParams(scene, exportInfo, sceneNames)
 
-	if scene is not None:
-		if entry is None:
-			sceneTable.append(SceneTableEntry("UNTITLED_SCENE", scene.name + "_scene", None, '0', str(scene.sceneTableEntry.drawConfig), '0'))
-		else:
-			entry.sceneType = entry.sceneType
-			entry.config = str(scene.sceneTableEntry.drawConfig)
+	if scene is None:
+		sceneDrawConfig = None
 	else:
-		if entry is not None:
-			sceneTable.remove(entry)
-	writeSceneTable(sceneTable, sceneFileData, start, end, exportPath)
\ No newline at end of file
+		sceneDrawConfig = scene.sceneTableEntry.drawConfig
+
+	# ``DEFINE_SCENE()`` parameters
+	sceneParams = [sceneName, sceneTitle, sceneID, sceneDrawConfig, sceneUnk10, sceneUnk12]
+
+	# check if it's a custom scene name
+	# sceneIndex can be None and ootSceneOption not "Custom",
+	# that means the selected scene has been removed from the table
+	# however if the scene variable is not None
+	# set it to "INSERT" because we need to insert the scene in the right place
+	if sceneIndex is None and bpy.context.scene.ootSceneOption == "Custom": 
+		mode = "CUSTOM"
+	elif sceneIndex is None and scene is not None:
+		mode = "INSERT"
+	elif sceneIndex is not None:
+		mode = "NORMAL"
+	else:
+		mode = None
+
+	if mode is not None:
+		# if so, check if the custom scene already exists in the data
+		# if it already exists set mode to NORMAL to consider it like a normal scene
+		if mode == "CUSTOM":
+			exportName = bpy.context.scene.ootSceneName.lower()
+			for i in range(len(fileData)):
+				if fileData[i][0] == exportName + "_scene":
+					sceneIndex = i
+					mode = "NORMAL"
+					break
+		else:
+			exportName = exportInfo.name
+
+		# edit the current data or append new one if we are in a ``Custom`` context
+		if mode == "NORMAL":
+			for i in range(6):
+				if sceneParams[i] is not None and fileData[sceneIndex][i] != sceneParams[i]:
+					fileData[sceneIndex][i] = sceneParams[i]
+		elif mode == "CUSTOM":
+			sceneNames.append(sceneParams[2])
+			fileData.append(sceneParams)
+			sceneIndex = len(fileData) - 1
+		elif mode == "INSERT":
+			# if this the user chose a vanilla scene, removed it and want to export
+			# insert the data in the normal location
+			# shifted index = vanilla index - (vanilla last scene index - new last scene index)
+			index = getInsertionIndex(sceneNames, sceneID, None, mode)
+			sceneNames.insert(index, sceneParams[2])
+			fileData.insert(index, sceneParams)
+
+	# remove the scene data if scene is None (`Remove Scene` button)
+	if scene is None:
+		if sceneIndex is not None and sceneIndex < len(fileData):
+			if (exportName + "_scene") == fileData[sceneIndex][0]:
+				sceneNames.pop(sceneIndex)
+				fileData.pop(sceneIndex)
+			else:
+				raise PluginError("ERROR: Scene not found in ``scene_table.h``!")
```

Comment:
so initially I suggested the `< len()` check because of the except IndexError but actually it was only meant to check for None? should probably get rid of the len check as it's misleading

Also is the `else: raise` on the right indentation level? shouldn't it also be raised if `sceneIndex is None`?

Reply:
I'm not sure to understand, the len check makes sense since we're dealing with a list, it's just an extra safety check even though if should never be OoB

For the ``else: raise`` it makes more sense this way to me since it's with ``if (exportName + "_scene") == fileData[sceneIndex][0]:`` that we're actually looking for the scene, otherwise I'd prefer another error message as this one won't be really accurate since we never started checking the scene's existence